pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/nativefst/automaton/TransitionComparator.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.segment.local.utils.nativefst.automaton;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+
+class TransitionComparator implements Comparator<Transition>, Serializable {
```

Comment:
Is there more than one sort order for `Transition`? If not, I suggest making `Transition implement Comparable<Transition>`

Reply:
Transition is designed to really be pluggable -- users should be able to write their own versions of Transition without having to worry about comparisons unless they want a custom order. This class exists more from an extensibility perspective